gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -325,3 +1216,102 @@ nns_edge_get_topic (nns_edge_h edge_h, char **topic)
 
   return NNS_EDGE_ERROR_NONE;
 }
+
+/**
+ * @brief Set nnstreamer edge info
+ * @return 0 if OK, negative value if error
+ */
+int
+nns_edge_set_info (nns_edge_h edge_h, const char *key, const char *value)
+{
+  nns_edge_handle_s *eh;
+  char *ret_str = NULL;
+
+  eh = (nns_edge_handle_s *) edge_h;
+
+  if (!NNS_EDGE_MAGIC_IS_VALID (eh)) {
+    nns_edge_loge ("Invalid param, given edge handle is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (0 == g_ascii_strcasecmp (key, "CAPS")) {
+    ret_str = g_strdup_printf ("%s%s", eh->caps_str, value);
```

Comment:
Q: replace old value or append new string
IMO replacing caps string is better (user handles the capability of edge handle)

Reply:
Marked as todo item. Thanks.